### ISSUE TITLE
FUSE bug: rename isn't updating the parent of a moved child.

### ISSUE THREAD

**Issue by kaushiks** [osxfuse/osxfuse#466]
I'm writing a simple FUSE filesystem that is very similar to the loopback filesystem from the examples directory [1]. I believe I've run into what appears to be a bug in FUSE (that I'm able to reproduce in my filesystem and the loopback filesystem). 

The command I use to run the loopback filesystem is:
sudo ./loopback /Volumes/loop -omodules=threadid:subdir,subdir=/tmp/dir -oallow_other,native_xattr,volname=LoopbackFS

I then am able to reproduce the bug by running the following commands: 
```
cd /Volumes/loop
mkdir foo
cd foo
mkdir -p bar/baz
mkdir top
mv bar top
cd top/bar
echo 3 > ../../../file.txt
```

This says 
`
-bash: ../../../1: Permission denied`

because it is resolving ../../../1 to /Volumes/1.

The "mv bar top" command is what appears to confuse FUSE and I believe is corrupting internal state. Often, after I reproduce the bug I experience other "random" failures like the loopback filesystem not being able to cat a file that exists. 

Any help diagnosing this problem is appreciated!

[1] https://github.com/osxfuse/filesystems/blob/master/filesystems-c/loopback/loopback.c
 
Edit:
I'm on OSX 10.12.6 (16.7.0 Darwin Kernel Version 16.7.0: Mon Nov 13 21:56:25 PST 2017; root:xnu-3789.72.11~1/RELEASE_X86_64 x86_64)
I'm running v3.7.1 of the FUSE driver: com.github.osxfuse.filesystems.osxfuse (3.7.1) 8C89BD7B-DA6D-347D-AC76-759C45AE6099

**Comment by kaushiks** [comment]
FWIW, I did some debugging and this doesn't appear to be a bug in the user mode code. 

If I do this
```
mkdir foo
cd foo
mkdir -p bar/baz
mkdir top
mv bar top
cd top/bar/baz
echo test > ../../../file.txt
```
and then look at the FUSE callbacks that are getting called, I see a couple of getattr calls:
```
...
2051        0 getattr          0          /foo/top/bar
2052        0 getattr          0          /foo
...
```
Oddly, it skips over calling getattr for /foo/top. My theory is that this is because the driver is failing to recognize that bar's parent has changed after the rename. It appears to wrongly assume that foo is still bar's parent.

@bfleischer Any thoughts on how I can fix and/or work around this issue? Thanks!

**Comment by kaushiks** [comment]
https://groups.google.com/forum/#!searchin/osxfuse-group/mv$20barry%7Csort:date/osxfuse-group/XamzKH7BA30/yDPt68FsBQAJ is another instance of this problem that I'm able to reproduce in my filesystem too.

**Comment by kaushiks** [comment]
Hi @bfleischer, Could you take a look at this bug report please? Happy to provide any other context / code you might need to reproduce the problem. Thanks.

**Comment by bfleischer** [comment]
Thank you very much for looking into the issue and the pull request! The bug should be fixed in the soon to be released version 3.8.3.
